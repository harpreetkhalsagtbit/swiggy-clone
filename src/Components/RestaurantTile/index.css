.restaurant-tile {
    position: relative;
    width: 254px;
    flex: 1 0 30%; /* explanation below */
    margin-top: 30px;
    margin-bottom: 60px;
}

.restaurant-tile-wrapper {
    background: #fff;
    display: block;
    text-decoration: none;
    color: inherit;
    margin: -20px -20px -41px;
}

.restaurant-tile-body {
    position: relative;
    padding: 20px 20px 57px;
    border: 1px solid #fff;
    contain: content;

    width: 254px;
}

.restaurant-tile-body-image-wrapper {
    background: #eef0f5;
    width: 254px;
    height: 160px;
    position: relative;
}

.restaurant-tile-body-title-wrapper {
    margin-top: 14px;
}

.restaurant-tile-body-title {
    font-size: 17px;
    font-weight: 500;
    word-break: break-word;
}

.restaurant-tile-body-subtitle {
    color: #686b78;
    font-size: 13px;
    margin-top: 4px;
}

.restaurant-tile-body-desc-wrapper {
    display: flex;
    align-items: center;
    
    margin-top: 18px;
    font-size: 12px;
    -ms-flex-pack: justify;
    justify-content: space-between;
    color: #535665;
}

.restaurant-tile-body-desc-ratings {
    display: flex;
    align-items: center;

    background-color: #48c479;
    color: #fff;

    height: 20px;
    width: 43px;
    padding: 0 5px;
    font-weight: 400;
}

.star {
    font-size: 15px;
    font-weight: bold;
    margin-right: 4px;
    position: relative;
    top: -1px;
}

[class^=icon-] {
    font-family: icomoon;
    speak: none;
    font-style: normal;
    font-weight: 400;
    font-variant: normal;
    text-transform: none;
    line-height: 1;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}

.restaurant-tile-body-desc-ratings .icon-star{
    font-size: 10px;
    margin-right: 4px;
    position: relative;
    top: -1px;
}

.restaurant-tile-body-quick-view {
    color: #686b78;
    font-size: 13px;
    visibility: hidden;
    border-top: 1px solid #e9e9eb;
    padding-top: 14px;
    margin-top: 14px;
    position: absolute;
    left: 20px;
    right: 20px;
    bottom: 14px;
}

.restaurant-tile-body-quick-view-title {
    text-align: center;
    float: none;
    display: block;
    margin: 0 auto;
    width: 100px;
    color: #5d8ed5;
    /* float: right; */
    text-transform: uppercase;
    font-weight: 600;
}

.restaurant-tile-body:hover {
    border-color: #d3d5df;
    box-shadow: 0 4px 7px 0 rgba(218,220,230,.6);
}

.restaurant-tile-body:hover .restaurant-tile-body-quick-view {
    visibility: visible;
}