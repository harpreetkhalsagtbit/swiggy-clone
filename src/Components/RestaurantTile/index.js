import React from "react";

import "./index.css";

const RestaurantTile = ({ restaurant }) => {
    return (
        <div className="restaurant-tile">
            <a href="#" className="restaurant-tile-wrapper">
                <div className="restaurant-tile-body">
                    <div className="">
                        <div className="restaurant-tile-body-image-wrapper"><img className="" alt="Frozen Bottle" width="254" height="160" src="https://res.cloudinary.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_508,h_320,c_fill/b6bsido84405494xesjy" /></div>
                        <div className="restaurant-tile-body-title-wrapper">
                            <div className="restaurant-tile-body-title">{restaurant.name}</div>
                            <div className="restaurant-tile-body-subtitle" title={restaurant.food_types.join(', ')}>{restaurant.food_types.join(', ')}</div>
                        </div>
                        <div className="restaurant-tile-body-desc-wrapper">
                            <div className="restaurant-tile-body-desc-ratings"><span className="star">&#9734;</span><span>{restaurant.ratings || "N/A"}</span></div>
                            <div>•</div>
                            <div>{restaurant.delivery_time}</div>
                            <div>•</div>
                            <div className="">₹{restaurant.price_for_two} FOR TWO</div>
                        </div>
                        <div className="restaurant-tile-body-quick-view">
                            <span role="button" aria-label="Open" className="restaurant-tile-body-quick-view-title">Quick View</span>
                        </div>
                    </div>
                </div>
            </a>
        </div>
    )
};
export default RestaurantTile;
