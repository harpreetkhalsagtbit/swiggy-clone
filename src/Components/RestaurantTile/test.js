import React from 'react'
import {render, fireEvent} from '@testing-library/react'
import RestaurantTile from '../RestaurantTile/index.js'

describe('<RestaurantTile /> spec', () => {
    let restaurant = 			{
        "name": "Hot Spot Mini Shop",
        "food_types": ["Fast Food", "Snacks", "North Indian", "Chinese", "Italian", "Desserts", "Beverages"],
        "ratings": "4.0",
        "isExlusive": true,
        "delivery_time": "108 mins",
        "price_for_two": 1300
    }
    it('renders the component', () => {

        let container = render(<RestaurantTile restaurant={restaurant}/>)
        expect(container.firstChild).toMatchSnapshot()
    })
    it('assert Show More Tiles displays correct data', () => {
        render(<RestaurantTile restaurant={restaurant}/>)
 
        expect(document.querySelector('.restaurant-tile-body-title').innerHTML).toBe("Hot Spot Mini Shop")
        expect(document.querySelector('.restaurant-tile-body-subtitle').innerHTML).toBe("Fast Food, Snacks, North Indian, Chinese, Italian, Desserts, Beverages")
        expect(document.querySelector('.restaurant-tile-body-desc-wrapper').innerHTML).toBe('<div class="restaurant-tile-body-desc-ratings"><span class="star">☆</span><span>4.0</span></div><div>•</div><div>108 mins</div><div>•</div><div class="">₹1300 FOR TWO</div>')
    })
})
