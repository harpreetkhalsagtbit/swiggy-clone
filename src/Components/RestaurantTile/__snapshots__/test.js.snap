// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantTile /> spec renders the component 1`] = `undefined`;
